src/auth/user.repisitory.ts

```diff
@@ -0,0 +1,103 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { UserEntity } from './user.entity';
+import { AuthDto } from './dto/auth.dto';
+import { comparePasswordHash, getPasswordHash } from 'src/helpers/password-hesh';
+
+@Injectable()
+export class UserRepository {
+    constructor(
+        @InjectRepository(UserEntity)
+         private readonly userRepository: Repository<UserEntity>,
+    ) {}
+
+    async createAndSaveUser(user: AuthDto): Promise<UserEntity> {
+        const password = user.password;
+        const passwordHash = await getPasswordHash(password);
+
+        const foundUsers = await this.userRepository.find(
+           { where: {
+                name: user.name,
+                email: user.email,
+            }
+        });
+
+        for (const foundUser of foundUsers) {
+            const res = await comparePasswordHash(password, foundUser.passwordHash);
+            if (res) {
+                throw new BadRequestException('User already exists');
+            }
+        }     
+        
```

Comment:
why we check password hashes here? we don't need it. (if user already registered with this email why should throw error about user existence without password checking)
we don't need a user existence checking in repository. It should be moved into service (it's a part of business logic, not a part of data storing logic)